docs/apache-airflow/security/access-control.rst

```diff
@@ -37,101 +37,113 @@ regarding its security model.
-There is a special view called ``DAGs`` (it was called ``all_dags`` in versions 1.10.*) which
-allows the role to access all the dags. The default ``Admin``, ``Viewer``, ``User``, ``Op`` roles can all access ``DAGs`` view.
+Creating custom roles is the recommended way to customize your environment access. To do so:
 
-.. image:: /img/add-role.png
-.. image:: /img/new-role.png
+1. In the **List Roles** menu, click the blue button to create a new role. Alternatively, select an existing role and click **Actions** > **Copy Role** to create a role based on your selection.
 
-The image shows the creation of a role which can only write to
-``example_python_operator``. You can also create roles via the CLI
-using the ``airflow roles create`` command, e.g.:
+ .. image:: /img/list-roles.png
 
-.. code-block:: bash
+2. Specify a name and add permissions for the role. Note that the names for permissions will vary slightly between the UI and the source code.
 
-  airflow roles create Role1 Role2
+ .. image:: /img/add-permissions.png
 
-And we could assign the given role to a new user using the ``airflow
-users add-role`` CLI command.
+3. Click Save.
 
+You can also create roles via the CLI using the following command:
 
-Permissions
-'''''''''''
+.. code-block:: bash
+
+  airflow roles create <Role1> <Role2>
+
+You could then assign the given role to a new user using the ``airflow
+users add-role`` CLI command. Note that adding and removing permissions for a role can only be completed in the UI.
 
 Resource-Based permissions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Starting with version 2.0, permissions are based on individual resources and a small subset of actions on those
+Starting with version 2.0, permissions are based on individual resources and a small set of actions with those
 resources. Resources match standard Airflow concepts, such as ``Dag``, ``DagRun``, ``Task``, and
 ``Connection``. Actions include ``can_create``, ``can_read``, ``can_edit``, and ``can_delete``.
 
-Permissions (each consistent of a resource + action pair) are then added to roles.
+Each endpoint in Airflow requires one or more permissions to be accessed. To access an endpoint, a role needs all permissions required by that endpoint.
 
-**To access an endpoint, the user needs all permissions assigned to that endpoint**
+DAG-Based Permissions
+^^^^^^^^^^^^^^^^^^^^^
+``Admin`` users can create roles with permission to access only specific DAGs. This is called DAG level access.
 
-There are five default roles: Public, Viewer, User, Op, and Admin. Each one has the permissions of the preceding role, as well as additional permissions.
+While access to all DAGs is controlled with resource-based permissions, DAG level access only has two associated permissions:
+``can_read`` and ``can_edit`` (``can_dag_read`` and ``can_dag_edit`` were deprecated in 2.0.0).
```

Comment:
This is a bit confusing. Something along the lines of "Any resource-based permission applies to all DAGs, but it's also possible to assign `can_read` and `can_edit` to individual DAGs. (``can_dag_read`` and ``can_dag_edit`` were deprecated in 2.0.0). This was added because a lot of users wanted the ability to control who could read or run individual DAGs."

Reply:
@jhtimmins Tried to clarify this by making a list of differences